### DISCUSSION TITLE
Migrate from 8 to 9

### DISCUSSION THREAD

**Discussion by ScreamZ** [pnpm/pnpm#8003] [General]
Hello @zkochan 

i'm not using corepack and so far I'm using `volta install pnpm` to manage my projects.

```
~/dev-workspace/pro/supafriends/turbo-mono (main*) » pn dev                                                                                                            screamz@Scr34mZ
 ERR_PNPM_BAD_PM_VERSION  This project is configured to use v8.9.0 of pnpm. Your current pnpm is v9.0.5
```

I've no reference to pnpm anywhere in the project, how I'm i supposed to update it to v9 ?

I don't want to get rid of the error message, i just want a good migration.
This is catastrophic to release this and not give any intel about of to proceed to manage project on old version along pnpm 9

How to proceed

**Comment by zkochan** [comment]
Well, your `package.json` has a `packageManager` field with `pnpm@8.9.0` specified in it. So, either install pnpm via corepack, which will automatically switch to the version of pnpm set in the `packageManager` field, or remove the field, or change the version of pnpm in that field, or set the [package-manager-strict](https://pnpm.io/npmrc#package-manager-strict) setting to `false`.

**Reply by ScreamZ** [reply]
I didn't had the key `packageManager` in my `package.json` that's why I didn't understand, maybe some caching mechanism inside pnpm cache ?

I defined `packageManager` and it worked.

I'll try the others solutions, also corepack would be welcome. Never tested but if it's stable now, could be nice. I was relying on volta atm.

Thank you for your kind response (whereas mine wasn't much :x)

Have a nice day

**Reply by zkochan** [reply]
Maybe it was configured in the `package.json` in the root of the monorepo? If it fails without the field, then it is a bug but provide some sample repo that reproduces the issue.
